content/articles/object-based-image-analysis-using-matlab/index.md

```diff
@@ -0,0 +1,235 @@
+
+First, we import the image using the following codes.
+```matlab
+i = imread('print.PNG'); %importing the image
+figure,imshow(i) %  Displaying the imported image
+```
+![The imported image](/engineering-education/object-based-image-analysis-using-matlab/objectanalysis_one.jpg)
+
+The imported image is then converted to a grayscale image using the codes below.
+```matlab
+i2 = rgb2gray(i) %coverting the imported RGB image to grayscale image
+figure,imshow(i2)% Displaying the grayscale image 
+```
+![Grayscale image](/engineering-education/object-based-image-analysis-using-matlab/objectanalysis_two.jpg)
+
+We will use the grayscale image `i2` to demonstrate various edge detection methods in images. The following are methods used in edge detection:
+
+- Sobel; in this method, the edge is determined from the points with the highest gradient. This method is executed using the function `sobel()` and the syntax is `i_edge = edge(i_gray,'sobel')`.
+ The below codes demonstrates edge detection method using the sobel method.
+```matlab
+s = edge(i2,'sobel'); % edge detection using sobel method
+figure,imshow(s) %displaying edges detected using sobel method
+```
+![Edges detected using sobel method](/engineering-education/object-based-image-analysis-using-matlab/objectanalysis_three.jpg)
+
+- Prewitt: This is a suitable method of estimating the magnitude and orientation of edges in an image. It gives a more detailed output compared to the sobel method. The output image may be noisy depending on the number of edges present and their proximity. This method is executed using the function `prewitt()` and the syntax is `i_edge = edge(i_gray,'prewitt')`.
+ The below codes shows edge detection using prewitt method.
+```matlab
+p = edge(i2,'prewitt'); %edge detection using prewitt method
+figure, imshow(p) %displaying edges detected using prewitt method
+```
+![Edges detected using Prewitt method](/engineering-education/object-based-image-analysis-using-matlab/objectanalysis_four.jpg)
+
+- Log: This method smoothens the image then executes the laplacian function resulting in a double-edged image. This method uses the function `log()` and the syntax is `i_edge = edge(i_gray,'log')`.
+ The below codes shows edge detection using log method.
+```matlab
+l = edge(i2,'log'); %edge detection using log method
+figure, imshow(l) %displaying edges detected using log method
+```
+![Edge detected using log method](/engineering-education/object-based-image-analysis-using-matlab/objectanalysis_five.jpg)
+
+- Canny: This method detects the edges by separating noise from the image. This is better because it does not disturb the features of the edges in the image. This method uses the function `canny()` and the syntax is `i_edges = edge(i_gray,'canny')`.
+ The below codes shows edge detection using the canny method.
+```matlab
+c = edge(i2,'canny'); %edge detection using canny method
+figure, imshow(c) %displaying edges detected using canny method
+```
```

Comment:
Hello, what does code do? why is the `edge(i2,'canny')`, described this way? I've requested this severally @alpho361!